From e4a324ce1f4ffca214b6d5b5681b9eca630707bf Mon Sep 17 00:00:00 2001
From: Laurent Vivier <lvivier@redhat.com>
Date: Tue, 29 Jul 2025 17:00:50 +0200
Subject: [PATCH] Declare rtl8139 as deprecated
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

RH-Author: Laurent Vivier <lvivier@redhat.com>
RH-MergeRequest: 396: Declare rtl8139 as deprecated
RH-Jira: RHEL-45624
RH-Acked-by: Eugenio Pérez <eperezma@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Cindy Lu <lulu@redhat.com>
RH-Commit: [1/1] d44c1b68b59f1442b01c84b43e07aed5f4d254bc (lvivier/qemu-kvm-centos)

JIRA: https://issues.redhat.com/browse/RHEL-45624
Upstream: RHEL ONLY

The e1000+e1000e pair should be more than sufficient for the
"ancient non-virtio NIC" use case.

There does not appear to be any compelling reason to continue supporting
the rtl8139 NIC, so we should deprecate it in RHEL-10, with a view to
deleting it in RHEL-11.

Signed-off-by: Laurent Vivier <lvivier@redhat.com>
---
 hw/net/rtl8139.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/hw/net/rtl8139.c b/hw/net/rtl8139.c
index 31a6956252..7609006394 100644
--- a/hw/net/rtl8139.c
+++ b/hw/net/rtl8139.c
@@ -57,6 +57,7 @@
 #include "system/dma.h"
 #include "qemu/module.h"
 #include "qemu/timer.h"
+#include "qemu/error-report.h"
 #include "net/net.h"
 #include "net/eth.h"
 #include "system/system.h"
@@ -3364,6 +3365,9 @@ static void pci_rtl8139_realize(PCIDevice *dev, Error **errp)
     DeviceState *d = DEVICE(dev);
     uint8_t *pci_conf;
 
+    warn_report("'rtl8139' is deprecated, "
+                "please use a different Network Interface Card");
+
     pci_conf = dev->config;
     pci_conf[PCI_INTERRUPT_PIN] = 1;    /* interrupt pin A */
     /* TODO: start of capability list, but no capability
-- 
2.39.3

